cli/src/pcluster/validators/awsbatch_validators.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pcluster.models.common import DynamicParam, FailureLevel, Param, Validator
+from pcluster.utils import get_supported_architectures_for_instance_type, is_instance_type_format
+
+
+class AwsbatchComputeResourceValidator(Validator):
+    """Awsbatch compute resource validator."""
+
+    def _validate(self, min_vcpus: Param, desired_vcpus: Param, max_vcpus: Param):
+        """Validate min, desired and max vcpus combination."""
+
+        if desired_vcpus.value < min_vcpus.value:
+            self._add_failure(
+                "The number of desired vcpus must be greater than or equal to min vcpus", FailureLevel.ERROR
+            )
```

Comment:
Minor: You are not marking the any param as invalid. If you want, you can write
```
self._add_failure(
                "The number of desired vcpus must be greater than or equal to min vcpus", FailureLevel.ERROR, [desired_vcpus, min_vcpus]
            )
```

Reply:
But if the validators are not as intertwined as EBS or FSx validator, keeping the current code is fine.

Reply:
Fixed